core/kernel/transfer_list.c

```diff
@@ -389,14 +394,26 @@ bool transfer_list_set_data_size(struct transfer_list_header *tl,
 		    tl->size + mov_dis > tl->max_size) {
 			return false;
 		}
-		ru_new_ev = old_ev + mov_dis;
-		memmove((void *)ru_new_ev, (void *)old_ev, tl_old_ev - old_ev);
+		r_new_ev = old_ev + mov_dis;
 		tl->size += mov_dis;
-		gap = ru_new_ev - new_ev;
 	} else {
-		gap = old_ev - new_ev;
+		/*
+		 * Move distance should be round down to meet the requirement of
```

Comment:
How about:
`/* Move distance should be rounded down to match the data alignment */`

Reply:
Fixed. I updated the comment for the condition of `rounded up` as well.